warehouse/query-core/src/main/java/datawave/query/iterator/builder/IvaratorBuilder.java

```diff
@@ -44,10 +45,20 @@ protected void validateIvaratorControlDir(IvaratorCacheDir ivaratorCacheDir) {
         String ivaratorCacheDirURI = ivaratorCacheDir.getPathURI();
         FileSystem hdfsFileSystem = ivaratorCacheDir.getFs();
         
-        final URI hdfsCacheURI;
         try {
-            hdfsCacheURI = new URI(ivaratorCacheDirURI);
-            hdfsFileSystem.mkdirs(new Path(hdfsCacheURI));
+            final Path hdfsCachePath = new Path(new URI(ivaratorCacheDirURI));
+            // get the parent directory
+            final Path parentCachePath = hdfsCachePath.getParent();
+            final URI parentURI = parentCachePath.toUri();
+            if (!hdfsFileSystem.exists(parentCachePath)) {
+                // being able to make the parent directory is proof enough
+                hdfsFileSystem.mkdirs(parentCachePath);
+            } else if (parentURI.getScheme().equals("file")) {
+                File parent = new File(parentURI.getPath());
+                if (!parent.canWrite() || !parent.canRead()) {
+                    throw new IllegalStateException("Invalid permissions to directory " + parentCachePath);
+                }
+            }
```

Comment:
There's the case where the object exists on hdfs, but is not a directory. It's unlikely, but consider double-checking here?

Reply:
done